GIT-BRANCHING-STRATEGY.md

```diff
@@ -0,0 +1,123 @@
+
+1. Create a release branch from the develop branch as you get close to your release or other milestones, such as the end of the sprint.
+2. Give this branch a clear name associating it with a release, for example `release/1.0.0`.
+
+* May branch off from: `develop`
+* Must merge back into: `develop` and `master`
+* Branch naming convention: `release/*`
+* The `develop` branch is cleared to receive features for the next big release
+* The key moment to branch off a new release branch from `develop` is when `develop` (almost) reflects the desired state of the new release.
+* It is exactly at the start of a release branch that the upcoming release gets assigned a version number—not any earlier.
+
+### Hotfix branches
+
+* May branch off from: `master`
+* Must merge back into: `develop` and `master`
+* Branch naming convention: `hotfix/*`
+* Like release branches, but they arise from the necessity to act immediately upon an undesired state of a live production version
+
+### Key Stages Summary
+
+1. The repo is created with only a `master` branch, by default.
+2. A `develop` branch is created from `master`.
+3. `feature/*` branches are created from `develop`.
+4. When a feature is complete, it's merged into `develop` (via PR) and then removed.
+5. To initiate a release, a `release/*` branch is created from `develop`.
+6. When a release is complete, `release/*` is merged into `develop` and `master`, tagged, and then removed.
+7. If an issue in `master` needs to be resolved, a `hotfix/*` branch is created from `master`.
+8. When the hotfix release is complete, `hotfix/*` is merged into `develop` and `master`, tagged, and then removed.
+
+## Commits
+
+**Commit early, commit often!** Try to remember to push any pending commits to the remote repo at the end of every day so that all in-progress work is backed up.
+
+Commit messages should be detailed and helpful - avoid anything that's not a complete sentence. You should aim to tell a story in the commit message (i.e. what was broken and how it was fixed). A good rule of thumb to follow is to begin commit messages with a verb so that the message completes the phrase "This commit...".
+
+## Rebasing
+
+In order to improve the readability of the history, before submitting your work try to squash and rebase your git commits.
+
+1. Make sure you have the most up-to-date version of the branch you're rebasing on.
+2. Rebase you local branch. (solve any confilcts)
+3. Share your branch and create pull request.
+
+> **NB! You should avoid rebase branch that is already shared.**
+
+## Merging
+
+All merges into `develop`, `release/*`, and `hotfix/*` should happen via pull requests. This ensures that all code gets reviewed at some point before it's shipped. This ensures that all code gets reviewed at some point before it's shipped. For more information, see [Code Review Standards](CODE-REVIEW.md).
```

Comment:
Instead of copying `CODE-REVIEW.md` in this repo, wouldn't it be better to just [provide a link](https://github.com/thoughtbot/guides/tree/master/code-review)?